src/sbvr-api/errors.ts

```diff
@@ -6,9 +6,257 @@ export class PermissionParsingError extends TypedError {}
 export class UnsupportedMethodError extends TypedError {}
 export class SqlCompilationError extends TypedError {}
 export class SbvrValidationError extends TypedError {}
-export class InternalRequestError extends TypedError {}
 
 export class TranslationError extends TypedError {}
 export class ParsingError extends TypedError {}
-export class BadRequestError extends TypedError {}
-export class ForbiddenError extends TypedError {}
+
+export interface AnyObject {
+	[index: string]: any
+}
+
+export class HttpError extends TypedError {
+	body?: string | AnyObject
+
+	constructor(public status: number, error: string | Error = '') {
+		super(error)
+	}
+
+	setResponseBody(body: string | AnyObject) {
```

Comment:
Any reason for having a separate setter rather than including it in the constructor? How are you envisioning this being used?

Reply:
Like `throw new BadRequestError(err).setResponseBody({})` but can definitely also add that as a third (optional) parameter in the constructor (but then we will probably also have to update the constructors of of all `TypedError`s).

Reply:
It would only need to update the constructors of the `HttpError`s here, and in most cases it would be a lot more fluid imo to just have the extra constructor param when calling

Reply:
ok I'm on it